core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -114,6 +194,12 @@ private void invalidate(TableIdentifier ident) {
     tableCache.invalidateAll(metadataTableIdentifiers(ident));
   }
 
+  private void onTableExpiration(TableIdentifier ident) {
```

Comment:
this method seems to be not related to what it is doing

Reply:
It does seem like now that the `CacheWriter` is its own class, this could just be moved into the `EXPIRED` case.

Reply:
Agreed. Moved it directly into the class and got rid of the function.